### PR TITLE
Spawn app target start in async process and auto open browser window when flask http ready

### PR DESCRIPTION
**Pull request by ryanmalone**
tickets:
- https://trello.com/c/1y9Fw0Pi/120-2-app-target-start-should-launch-remote-python-process-asynchronously
& 
- https://trello.com/c/NithqaKr/119-3-app-target-start-should-launch-browser-window

### CODE REVIEW

**Comment by ryanmalone** [comment]
@carnesen dud files removed.

File: src/spawner/spawner-base/run-foreground.ts [hunk 1]
```diff
@@ -1,13 +1,51 @@
...
+        console.log(`${data}`);
+        if (data.includes('http://0.0.0.0:5000/')) {
+          open('http://0.0.0.0:5000/');
+        }
+      });
+    }
+
+    if (childProcess.stderr) {
+      childProcess.stderr.on('data', (data: any) => {
+        console.log(`Error: ${data}`);
```
Comment (carnesen) [inline src/spawner/spawner-base/run-foreground.ts]: Shouldn't this `console.log` be `process.stderr.write` to properly preserve newlines in the output? Similar to the stdout handler above.

Comment (ryanmalone) [reply src/spawner/spawner-base/run-foreground.ts]: done

File: src/spawner/spawner-base/run-foregroundSync.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+export function runForegroundSync(cmd: Cmd) {
+  const out = spawnSync(cmd.exe, cmd.args || [], {
+    cwd: cmd.cwd,
+    stdio: 'inherit',
+  });
+  if (out.error) {
+    throw out.error;
+  }
+}
```
[the remark this answers is not in the document]
Comment (ryanmalone) [reply src/spawner/spawner-base/run-foregroundSync.ts]: fixed

File: src/spawner/spawner-base/run-foreground.ts [hunk 3]
```diff
@@ -1,13 +1,51 @@
...
+    stdio: ['inherit', 'pipe', 'pipe'],
+  });
+
+  // Kill child when parent exits
+  signalExit(() => {
+    childProcess.kill();
+    process.exit(1);
+  });
+
+  await new Promise<void>(() => {
```
[the remark this answers is not in the document]
Comment (ryanmalone) [reply src/spawner/spawner-base/run-foreground.ts]: noted

File: src/spawner/spawner-base/run-foreground.ts [hunk 4]
```diff
@@ -1,13 +1,51 @@
...
+        process.stderr.write(data);
+        if (data.includes('http://0.0.0.0:5000/')) {
+          open('http://0.0.0.0:5000/');
+        }
+      });
+    }
+
+    if (childProcess.stderr) {
+      childProcess.stderr.on('data', (data: any) => {
+        console.log(`Error: ${data}`);
```
Comment (carnesen) [inline src/spawner/spawner-base/run-foreground.ts]:
```suggestion
        process.stderr.write(data);
```

**Pull request merged**